Dev/Gitlab/gitlab_merge_requests.js

```diff
@@ -0,0 +1,433 @@
+
+function request(path) {
+  const httpTransport = require('https');
+  const responseEncoding = 'utf8';
+  const httpOptions = {
+    hostname: gitlab_domain,
+    port: '443',
+    path,
+    method: 'GET',
+    headers: { 'PRIVATE-TOKEN': private_token }
+  };
+  httpOptions.headers['User-Agent'] =
+    'bitbar/gitlab_projects - node ' + process.version;
+
+  return new Promise((resolve, reject) => {
+    const request = httpTransport
+      .request(httpOptions, (res) => {
+        let responseBufs = [];
+        let responseStr = '';
+
+        res
+          .on('data', (chunk) => {
+            if (Buffer.isBuffer(chunk)) {
+              responseBufs.push(chunk);
+            } else {
+              responseStr = responseStr + chunk;
+            }
+          })
+          .on('end', () => {
+            responseStr =
+              responseBufs.length > 0
+                ? Buffer.concat(responseBufs).toString(responseEncoding)
+                : responseStr;
+
+            resolve(JSON.parse(responseStr));
+          });
+      })
+      .setTimeout(0)
+      .on('error', (error) => {
+        reject(error);
+      });
+    request.write('');
+    request.end();
+  });
+}
+
+const cachedProjectNames = {};
+
+async function getMRs() {
+  if (!fs.existsSync(SETTINGS_FILE_PATH)) {
+    const myUser = await request('/api/v4/user');
+    fs.writeFileSync(
+      SETTINGS_FILE_PATH,
+      JSON.stringify({
+        userId: myUser.id
+      })
+    );
+  }
+  const { userId } = JSON.parse(fs.readFileSync(SETTINGS_FILE_PATH));
+
+  const createdMRs = await request(
+    '/api/v4/merge_requests?scope=created_by_me&state=opened'
+  );
+  const assignedMRs = await request(
+    '/api/v4/merge_requests?scope=assigned_to_me&state=opened'
+  );
+  const approvalMRs = await request(
+    `/api/v4/merge_requests?scope=all&state=opened&approver_ids[]=${userId}`
+  );
+
+  const allMRs = assignedMRs
+    .concat(approvalMRs, createdMRs)
+    .reduce((uniqueMRs, currentMR) => {
+      if (!uniqueMRs.find(({ id }) => id === currentMR.id)) {
+        uniqueMRs.push(currentMR);
+      }
+      return uniqueMRs;
+    }, []);
+
+  const bitbar = globalRequire('bitbar');
+
+  if (allMRs.length === 0) {
+    // Hide when no open MRs
+    return bitbar([
+      {
+        image: gitlabIconBase64,
+        text: `0 MR`,
+        color: bitbar.darkMode ? 'white' : 'black',
+        dropdown: true
+      }
+    ]);
+  }
+
+  const statusCount = {};
+  function incrementStatusCount(key) {
+    statusCount[key] = statusCount[key] || 0;
+    statusCount[key] += 1;
+  }
+
+  const mergeRequestsByProjectId = {};
+
+  await Promise.all(
+    allMRs.map(async (mergeRequest) => {
```

Comment:
line 203, col 16, 'async functions' is only available in ES8 (use 'esversion: 8').